install/install_utils.php

```diff
@@ -2169,3 +2169,66 @@ function enableInsideViewConnector()
     // $mapping is brought in from the mapping.php file above
     $source->saveMappingHook($mapping);
 }
+
+/**
+ * Gets the username of the user under which the PHP script is currently running
+ * Notes:
+ * - works on Windows and Linux, tries a variety of methods to accomodate different systems and hosting restrictions
+ * - on Windows, return full username in form DOMAIN\USER
+ * - returns empty string if failed
+*/
+
+function getRunningUser()
+{
+    // works on Windows and Linux, but might return null on systems that include exec in disabled_functions in php.ini (typical in shared hosting)
+    $runningUser = exec('whoami');
+
+    if ($runningUser == null) {  // matches null, false and ""
+        if (is_windows())
```

Comment:
@anothermouse this is _inside_ getRunningUser, so that would cause an endless recursion. This function uses both methods, whoami and the one you suggested, it's probably overkill but my idea was to cover as many different scenarios as possible (where one function works, but not the other, and vice-versa).